docs/dev-guide/contributing.md

```diff
@@ -0,0 +1,38 @@
+# Contributing to Metalhead.jl
+
+We welcome contributions from anyone to Metalhead.jl! Thank you for taking the time to make our ecosystem better.
+
+You can contribute by fixing bugs, adding new models, or adding pre-trained weights. If you aren't ready to write some code, but you think you found a bug or have a feature request, please [post an issue](https://github.com/FluxML/Metalhead.jl/issues/new/choose).
+
+Before continuing, make sure you read the [FluxML contributing guide](https://github.com/FluxML/Flux.jl/blob/master/CONTRIBUTING.md) for general guidelines and tips.
+
+## Fixing bugs
+
+To fix a bug in Metalhead.jl, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). It would be helpful to file an issue first so that we can confirm the bug.
+
+## Adding models
+
+To add a new model architecture to Metalhead.jl, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). Keep in mind a few guiding principles for how this package is designed:
+
+- reuse layers from Flux as much as possible (e.g. use `Parallel` before defining a `Bottleneck` struct)
+- adhere as closely as possible to a reference such as a published paper (i.e. the structure of your model should be intuitive based on the paper)
+- use generic functional builders (e.g. [`resnet`](#) is the core function that build "ResNet-like" models based on the principles in the paper)
+- use multiple dispatch to add convenience constructors that wrap your functional builder
+
+When in doubt, just open a PR! We are more than happy to help review your code to help it align with the rest of the library. After adding a model, you might consider adding some pre-trained weights (see below).
+
+## Adding pre-trained weights
+
+To add pre-trained weights for an existing model or new model, you can [open a PR](https://github.com/FluxML/Metalhead.jl/pulls). Below, we describe the steps you should follow to get there.
+
+1. Train your model or port the weights from another framework.
+2. Save the model using [BSON.jl](https://github.com/JuliaIO/BSON.jl) with `BSON.@save "modelname.bson" model`. It is important that your model is saved under the key `model`.
+3. Compress the saved model as a tarball using `tar -cvzf modelname.tar.gz modelname.bson`.
+4. Obtain the SHAs (see the [Pkg docs](https://pkgdocs.julialang.org/v1/artifacts/#Basic-Usage)). Edit the `Artifacts.toml` file in the Metalhead.jl repository and add entry for your model. You can leave the URL empty for now.
+5. Open a PR on Metalhead.jl. Be sure to ping a maintainer (e.g. `@darsnack`) to let us know that you are adding a pre-trained weight. We will create a model repository on HuggingFace if it does not already exist.
+6. Open a PR to the [corresponding HuggingFace repo](https://huggingface.co/FluxML). We will merge your PR.
```

Comment:
maybe this is clear to regular HuggingFace users but not to me. How do I open a PR? Should I follow these instructions https://huggingface.co/docs/transformers/model_sharing?

Reply:
Good point, updated with more details and links to HF docs.